src/handlers.rs

```diff
@@ -16,19 +17,29 @@ use std::fs;
 use std::path::Path;
 use std::time::UNIX_EPOCH;
 use std::time::{Duration, Instant};
-use tokio::prelude::future;
-use tokio::prelude::*;
-use tokio::timer::{Delay, Interval};
+use tokio::timer::Delay;
 
-type HandlerResult = DenoResult<libdeno::deno_buf>;
-type Handler =
-  fn(d: *const DenoC, base: msg::Base, builder: &mut FlatBufferBuilder)
-    -> HandlerResult;
+// Buf represents a byte array returned from a "DenoOp".
+// The message might be empty (which will be translated into a null object on
+// the javascript side) or it is a heap allocated opaque sequence of bytes.
+// Usually a flatbuffer message.
+type Buf = Option<Box<[u8]>>;
+
+// JS promises in Deno map onto a specific Future
+// which yields either a DenoError or a message.
+type DenoOp = Future<Item = Buf, Error = DenoError>;
```

Comment:
I sometimes wonder why some things are DenoThings and others are just things.
I think people that are reading this code are generally aware that they're looking at deno code, so no need to mention it. 
It's largely a matter of taste, and Über Geschmack lässt sich nicht streiten.
This type IMO could just be called `Op`. Short and to the point.

Reply:
Fixed. (Yeah I just do that for sanity's sake sometimes - because I'm not exactly sure what Tokio or crate symbols have been loaded into the current namespace. A few days ago we had a discussion where I proposed banning allowing people to "use" a namespace in Rust - I'd rather explicitly list the symbols... This is why.)